### DISCUSSION TITLE
game running "laggy and slow" compared to stable

### DISCUSSION THREAD

**Discussion by Tinchozz** [ppy/osu#15404] [Q&A] [answered]
[runtime.log](https://github.com/ppy/osu/files/7457908/runtime.log)
[performance.log](https://github.com/ppy/osu/files/7457909/performance.log)

Sorry for the lack of specifics, but i don't know how to explain it. The game overall runs fine, at pretty much the same framerate than stable, but at gameplay it feels significantly worse, there's a big difference in smoothness and responsiveness between both versions. Tried changing framerates, made sure i was in fullscreen, both single and multithreaded. It sort of feels like the game is running at a lower refresh rate than it should (Dual monitor setup, in case that's important)

**Comment by peppy** [comment, marked as answer]
Are you able to capture a video and trim it down to an area where you feel the impact the most? Make sure you have the frame graphs visible in the video (Ctrl+F11 twice).

**Reply by Tinchozz** [reply]
[With unlimited fps](https://youtu.be/WLG71_y2KhE)
[2x refresh rate](https://youtu.be/uJfVjGzNQdU)

**Reply by peppy** [reply]
Can you see the same issues occurring when you watch the video back on youtube?

**Reply by Tinchozz** [reply]
Other than stutters when i go in and out of song select, not really. the video encoding kills it. I compared them both again and there seems to be more screen tearing on lazer, specially with 2x refresh rate, maybe that's it? Idk.
Again, apologies if this sounds dumb

**Reply by peppy** [reply]
If you're seeing screen tearing in lazer but *not* in stable, it means that stable has likely not been running at exclusive fullscreen. Which means you should be getting much better input latency on lazer as a result.

Either way, you can resolve this by reducing your frame limiter further, or using vsync. If you don't mind the added latency, try switching to borderless, which will guarantee that you don't see any screen tearing.

**Reply by Tinchozz** [reply]
i worded myself really bad there, there's no tearing in stable, only in lazer with 2x refresh rate. Tried reinstalling fresh on my nvme instead of my sata ssd just in case and nothing.
It seems to be something weird in my part, since i ran into the same issue last time i tried lazer a year ago.
Hopefully is one of those things that get randomly fixed by itself by the time stable gets descontinued.

**Reply by peppy** [reply]
Yes I understand what you means, and my response was in response to that. Did you try borderless?

**Reply by Tinchozz** [reply]
Yes, and all it does is add input lag.

**Reply by peppy** [reply]
And it doesn't help with the smoothness? Sounds like maybe an issue with your display or graphics drivers not correctly vsyncing if so..

Does running windowed change anything?

**Reply by Tinchozz** [reply]
oh yes, it definetly helps with smoothness, same with windowed.

**Reply by peppy** [reply]
So what you're seeing is just screen tearing then.

The solution to this is enabling something like gsync/freesync, or choosing a frame limiter that better maintains sync with your monitor. Preferrably the first.

**Reply by Tinchozz** [reply]
Ok this is highly unscientific but i think i figured out where the issue is. I tried turning off my custom resolution from 1080p75hz to the native 1080p60hz, and that made my stable feel as bad as lazer, while changing it in lazer made no difference whatsoever. Maybe lazer doesn't detect that my display runs at 75hz?

**Reply by peppy** [reply]
That's feasible. Are you "overclocking" your display? If so lazer may not be seeing it because it's not your hardware specs.

If your monitor is actually 75hz, could you please provide the model/manufacturer?

**Reply by Tinchozz** [reply]
Yes, i'm overcloking it to 75hz

**Reply by peppy** [reply]
You can track this here: https://github.com/ppy/osu-framework/issues/4081

**Comment by Saltssaumure** [comment]
I'm experiencing a similar problem - stable runs just fine, but lazer shows massive hitches in gameplay.
Here's a [60hz vsync recording](https://youtu.be/L7Dpcd3_4Vo). With higher/unlimited framerates, the result is about the same except with lots of screen tearing that doesn't show up in recording. Fullscreen might be slightly better than windowed, but it's not a huge difference either way.

Maybe relevant info:

- Also on a dual monitor setup
- [PC specs](https://uk.pcpartpicker.com/user/Saltssaumure/saved/hmM6RB)
- Using Windows 10, latest Nvidia driver
- No other tasks heavily using the GPU or CPU

**Reply by peppy** [reply]
I'm confused. You're recording that windowed with vsync? vsync is only valid when running exclusive fullscreen, so it's not a great test (we probably want to hide/disable/override it when running windowed).

Anyways, you're getting some very weird CPU spikes. Do you have much running in the background while lazer is running? Can you try without anything running and see if it helps?

**Reply by smoogipoo** [reply]
Were you importing beatmaps/skins/scores at the same time? Based on the frequency of GCs (the yellow boxes), this would be an indication.

**Reply by Saltssaumure** [reply]
@smoogipoo That's likely it actually, I didn't realise it would have that much of an effect on gameplay. It would be great if there was an option to auto-pause the import while playing. Temporarily pausing it in the notifications is a little scary since I don't want to accidentally cancel a multi-day process (I have every ranked map up to 2021, and writing to HDD takes a while).
